### PR TITLE
feat: add PTv3 multi-task segmentation and detection pipeline & transforms & cleanup

### PR DESCRIPTION
**Pull request by amadeuszsz**
## Summary

- Add a joint PTv3 segmentation + detection path with `autoware_ml.models.multi.PTv3SegDetModel`.
- Add multi-task PTv3 configs for NuScenes and T4Dataset, including tiny and pseudo-label variants.
- Add multiview detection datasets/datamodules for NuScenes and T4Dataset, with camera-order filtering and resolved lidar/image/sweep metadata.
- Add combined segmentation+detection datamodules for NuScenes and T4Dataset.
- Add vectorized PyTorch hard voxelization and PointPillars preprocessing utilities.
- Refactor PTv3 detection configs to use dataset-level detection metadata, unified metrics, and renamed 51 m NuScenes config.
- Update FRNet and PTv3 segmentation configs for the new dataset/config structure.
- Update calibration-status configs and datamodule handling.
- Simplify detection metric GT filtering to use range caps and `min_num_points`; remove attribute filtering from metric accumulation.
- Update NuScenes info generation to emit unified v1.1-style records, canonical detection categories, camera timestamp-aware `lidar2cam`, and shared per-frame records.
- Expand camera, camera-lidar, point-cloud, box, and segmentation transforms for multiview and joint-task pipelines.
- Add camera transform helpers for HWC/CHW/list image containers, plus GridMask and multiview normalization transforms.
- Normalize transform probability arguments to `p` and keep constructor docstrings aligned with transform signatures.
- Add tests for multiview datamodules, PTv3 segdet export, detection preprocessing, transform behavior, metrics, deployment, and MLflow config logging.
- Update PTv3, CLI, deployment, and MLflow docs for renamed configs, joint-task configs, and current export/deploy examples.

### Details

- New multi-task model:
  - Shares the PTv3 backbone between segmentation and detection heads.
  - Supports joint training/evaluation outputs for segmentation and 3D detection.
  - Adds export coverage for the combined PTv3 segdet model.

- Dataset and datamodule changes:
  - Add reusable multiview detection dataset/collation utilities.
  - Add NuScenes and T4Dataset multiview detection datamodules.
  - Add NuScenes and T4Dataset segmentation+detection datamodules.
  - Resolve camera image paths, lidar paths, sweeps, ego poses, and label/category mappings in dataset records.

- Detection and preprocessing:
  - Add vectorized hard voxelization in `autoware_ml.ops.voxelization`.
  - Add PointPillars-style detection preprocessing.
  - Update detection tests and fixtures around the new preprocessing and PTv3 detection flow.

- Transform updates:
  - Add multiview image loading, normalization, padding, resizing, crop/flip/rotate, and GridMask support for stacked/list image containers.
  - Add camera-lidar geometry updates that keep `lidar2img` aligned after global transforms/flips.
  - Add image augmentation matrix output for BEV/multiview pipelines.
  - Move point-cloud and segmentation augmentation probability parameters to `p`.
  - Keep transforms that create required downstream keys deterministic rather than probability-gated.
  - Refresh transform docstrings so constructor args are documented in `__init__`.

- NuScenes generation:
  - Build unified per-frame records with `images`, `lidar_points`, `instances`, pose fields, scene metadata, and optional segmentation masks.
  - Use canonical NuScenes detection categories and preserve ignored instance labels as `-1`.
  - Compute camera `lidar2cam` through camera and lidar ego poses to account for timestamp differences.

- Metrics and logging:
  - Rename detection metric minimum point filtering to `min_num_points`.
  - Remove detection metric attribute filtering and related distributed state handling.
  - Sanitize MLflow parameter keys before logger/hyperparameter logging.

- Documentation:
  - Update PTv3 docs with current segmentation, detection, and multi-task config names.
  - Update CLI/deployment examples for renamed configs and multiple weights.
  - Document MLflow/deployment behavior touched by the config/logging updates.

### Tests

- Added/updated datamodule tests for calibration status, detection3d, multiview detection3d, and segmentation3d.
- Added PTv3 segdet export tests.
- Added detection preprocessing tests.
- Updated transform tests for camera, camera-lidar, point-cloud, boxes3d, and segmentation3d changes.
- Updated detection metric, deployment, and MLflow utility tests.

## Checklist

- [ ] You've checked our [contribution overview](https://tier4.github.io/autoware-ml/contributing/contribution-overview/).
- [ ] Your PR follows our [pull request guidelines](https://tier4.github.io/autoware-ml/contributing/pull-request-guidelines/).

### CODE REVIEW

File: autoware_ml/transforms/camera/masking.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        self.p = p
+        self.ratio = ratio
+        self.rotate = rotate
+
+    def transform(self, input_dict: dict[str, Any]) -> dict[str, Any]:
+        """Mask images with a regular grid pattern."""
+        images, format_info = as_hwc_image_list(input_dict["img"])
+        masked = [self._grid_mask(image) for image in images]
+        input_dict["img"] = restore_image_container(input_dict["img"], masked, format_info)
+        return input_dict
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/transforms/camera/masking.py:37]: Irrelevant, handled by base class

File: autoware_ml/preprocessing/detection3d/point_pillar.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            Updated batch dictionary with the following additional keys:
+
+            - ``"voxels"`` - padded pillar features ``(total_pillars, max_num_points, C)``.
+            - ``"num_points"`` - per-pillar point counts ``(total_pillars,)``.
+            - ``"voxel_coords"`` - pillar coordinates ``(total_pillars, 4)`` in
+              ``[batch, z, y, x]`` order, ``dtype=torch.int32``.
+        """
+        device = batch_inputs_dict["points"][0].device
+        voxel_size = self.voxel_size.to(device=device)
+        point_cloud_range = self.point_cloud_range.to(device=device)
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/preprocessing/detection3d/point_pillar.py:87]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/utils/mlflow_helpers.py [hunk 3]
```diff
@@ -219,6 +226,47 @@ def should_enable_logger(cfg: DictConfig) -> bool:
...
+
+    Returns:
+        Sanitized MLflow parameter key.
+    """
+    sanitized_key = key
+    for source, target in MLFLOW_PARAM_KEY_REPLACEMENTS:
+        sanitized_key = sanitized_key.replace(source, target)
+    if MLFLOW_PARAM_KEY_PATTERN.fullmatch(sanitized_key) is None:
+        raise ValueError(f"Invalid MLflow parameter key after sanitization: {key}")
+    return sanitized_key
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/utils/mlflow_helpers.py:245]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/tools/dataset/nuscenes/generator.py [hunk 4]
```diff
@@ -296,40 +372,42 @@ def generate(
         task_generators = [create_task(task_name) for task_name in tasks]
 
         train_nusc_infos, val_nusc_infos = self._fill_trainval_infos(
-            nusc, train_scenes, val_scenes, test, max_sweeps, task_generators, root_path
+            nusc, train_scenes, val_scenes, test, task_generators, root_path
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/tools/dataset/nuscenes/generator.py]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/models/multi/ptv3_segdet.py [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+        return metrics
+
+    def build_eval_output(
+        self, batch: Mapping[str, Any], outputs: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Produce detection and segmentation eval data for both task metrics."""
+        eval_out = detection_eval_output(self.bbox_head.predict(outputs["det_outputs"]), batch)
+        eval_out["seg_pred_labels"] = outputs["seg_logits"].argmax(dim=1)
+        eval_out["seg_target_labels"] = batch["segment"].long()
+        eval_out["seg_coord"] = batch["coord"]
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/models/multi/ptv3_segdet.py]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/tools/dataset/nuscenes/generator.py [hunk 6]
```diff
@@ -37,6 +37,124 @@
...
+
+    instances = []
+    for i in range(len(gt_boxes)):
+        label = int(categories.get(str(names[i]), -1))
+        instances.append(
+            {
+                "bbox_label": label,
+                "bbox_3d": [float(value) for value in gt_boxes[i]],
+                "bbox_3d_isvalid": bool(valid_flag[i]),
+                "bbox_label_3d": label,
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/tools/dataset/nuscenes/generator.py:109]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/ops/voxelization/voxelization.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        flat_idx.unsqueeze(1).expand(-1, C),
+        valid_points,
+    )
+
+    # Count valid points per voxel (capped at max_num_points)
+    num_points = torch.zeros(M, dtype=torch.int32, device=device)
+    num_points.scatter_add_(
+        0,
+        valid_voxel_ids.to(torch.int32),
+        torch.ones(valid_voxel_ids.shape[0], dtype=torch.int32, device=device),
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/ops/voxelization/voxelization.py:143]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/utils/runtime.py [hunk 8]
```diff
@@ -167,4 +168,6 @@ def log_hyperparameters(cfg: DictConfig, trainer_logger: Logger | None, trainer:
     """
     if trainer_logger is None:
         return
-    trainer.logger.log_hyperparams(OmegaConf.to_container(cfg, resolve=True))
+
+    params = OmegaConf.to_container(cfg, resolve=True)
+    trainer.logger.log_hyperparams(sanitize_mlflow_param_keys(params))
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/utils/runtime.py]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/transforms/camera/resize.py [hunk 9]
```diff
@@ -77,3 +85,199 @@ def _update_camera_matrix(
...
+            intrinsics[camera_index, 0, 0] *= scale_x
+            intrinsics[camera_index, 1, 1] *= scale_y
+            intrinsics[camera_index, 0, 2] *= scale_x
+            intrinsics[camera_index, 1, 2] *= scale_y
+
+        input_dict["img"] = np.stack(resized_images, axis=0).astype(np.float32)
+        input_dict["camera_intrinsics"] = intrinsics
+        if "lidar2img" in input_dict:
+            input_dict["lidar2img"] = intrinsics @ input_dict["lidar2cam"]
+        return input_dict
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/transforms/camera/resize.py:130]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/889684f3708034c49c89b3ce1edce9e0dcc29bdf

File: autoware_ml/transforms/camera/resize.py [hunk 10]
```diff
@@ -77,3 +85,203 @@ def _update_camera_matrix(
...
+            intrinsics[camera_index, 1, 1] *= scale_y
+            intrinsics[camera_index, 0, 2] *= scale_x
+            intrinsics[camera_index, 1, 2] *= scale_y
+
+        input_dict["img"] = restore_image_container(input_dict["img"], resized_images, format_info)
+        input_dict["camera_intrinsics"] = intrinsics[0] if intrinsics_was_single else intrinsics
+        if "lidar2img" in input_dict:
+            lidar2cam = input_dict["lidar2cam"]
+            input_dict["lidar2img"] = input_dict["camera_intrinsics"] @ lidar2cam
+        return input_dict
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/transforms/camera/resize.py:130]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/1c0c369aec4c2d30c4f19db84f438ead621c1ed8

File: docs/contributing/adding-models.md [hunk 11]
```diff
@@ -378,14 +378,14 @@ Use these rules when creating `<variant>`:
 
 - include only future-distinguishing choices such as backbone, modality, voxel size, or range
 - do not encode properties that are inherent to the model family
-- normalize voxel sizes as `voxel020`, `voxel005`, `voxel030`
+- normalize voxel sizes as `voxel020`, `voxel005`, `voxel0170`
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply docs/contributing/adding-models.md]: Part of https://github.com/tier4/autoware-ml/pull/58/commits/1c0c369aec4c2d30c4f19db84f438ead621c1ed8
